src/Functions/Regexps.h

```diff
@@ -267,45 +268,91 @@ inline Regexps constructRegexps(const std::vector<String> & str_patterns, [[mayb
 
     /// If not HS_SUCCESS, it is guaranteed that the memory would not be allocated for scratch.
     if (err != HS_SUCCESS)
-        throw Exception("Could not allocate scratch space for hyperscan", ErrorCodes::CANNOT_ALLOCATE_MEMORY);
+        throw Exception(ErrorCodes::CANNOT_ALLOCATE_MEMORY, "Could not allocate scratch space for hyperscan");
 
     return {db, scratch};
 }
 
-/// If WithEditDistance is False, edit_distance must be nullopt
-/// Also, we use templates here because each instantiation of function
-/// template has its own copy of local static variables which must not be the same
-/// for different hyperscan compilations.
+/// Maps string pattern vectors + edit distance to compiled vectorscan regexps. Uses the same eviction mechanism as the LocalCacheTable for
+/// re2 patterns. Because vectorscan regexes are overall more heavy-weight (more expensive compilation, regexes can grow up to multiple
+/// MBs, usage of scratch space), 1. GlobalCacheTable is a global singleton and needs locking 2. the pattern compilation is done outside
+/// GlobalCacheTable's lock (at the cost of another level of locking).
+struct GlobalCacheTable
+{
+    constexpr static size_t CACHE_SIZE = 500; /// collision probability
```

Comment:
How was it picked?

Reply:
Picked the constant by some silly rule of thumb a la "the reasonable total cache size should not exceed 1 GB and each pattern has a worst case size of 2 MB" --> 500 patterns..